source/mainnet/smart-contracts/tutorials/wCCD/wCCD-interacting.rst

```diff
@@ -0,0 +1,752 @@
+not support the standard ``12345`` but it supports the ``CIS-0`` and the ``CIS-1`` standards.
+
+.. image:: ./images/wCCD_tutorial_9.png
+    :width: 100 %
+
+State-mutative functions
+------------------------
+
+The protocol has four state-mutative functions (``wrap``, ``unwrap``,
+``transfer``, and ``updateOperator``) that you can invoke on the ``proxy`` contract.
+These invokes will be passed through the fallback function on the ``proxy`` to the ``implementation`` contract.
+You require a different schema and JSON file with your input parameters for every invoke.
+All used schema files in this tutorial can be `downloaded <https://github.com/Concordium/concordium.github.io/tree/main/source/mainnet/smart-contracts/tutorials/wCCD/schemas>`_
+or you can create them as described in the
+comments of the `upgradable wCCD smart contract <https://github.com/Concordium/concordium-rust-smart-contracts/pull/128>`_.
+Because you will use a ``schema``,
+the input parameters can be provided with the ``--parameter-json`` flag.
+
+The ``wrap`` function
+=====================
+
+Wrapping CCD refers to the process of converting the native currency CCD into
+a ``CIS-2`` compliant token (wCCD) at a 1:1 ratio by sending CCD to the wCCD smart
+contract and getting wCCD in return.
+You can specify with the ``--amount`` flag how much CCD you want to wrap.
+
+Create a ``wrap.json`` file and insert the JSON object
+from option 1 (Receiver is an account) or option 2 (Receiver is a smart contract).
+
+.. dropdown:: Option 1 (Receiver is an account) (click here)
+
+    .. code-block::
+
+        {
+            "data": "",
+            "to": {
+                "Account": [
+                    ACCOUNT_ADDRESS
+                ]
+            }
+        }
+
+    The ``data`` field is only relevant if wCCD is sent to a smart contract as described in option 2.
+    You can use your account address if you want to credit the wCCD to your own account.
+    If you insert your account address correctly, the JSON object should look similar to the below JSON object.
+
+    .. code-block:: json
+
+        {
+            "data": "",
+            "to": {
+                "Account": [
+                    "4phD1qaS3U1nLrzJcgYyiPq1k8aV1wAjTjYVPE3JaqovViXS4j"
+                ]
+            }
+        }
+
+.. dropdown::  Option 2 (Receiver is a smart contract) (click here)
+
+    .. code-block::
+
+        {
+            "data": "",
+            "to": {
+                "Contract": [
+                    {
+                        "index": INDEX,
+                        "subindex": SUBINDEX
+                    },
+                    ENTRYPOINT_NAME
+                ]
+            }
+        }
+
+    The ``data`` field is relevant because wCCD is sent to a smart contract.
+    The ``OnReceivingCis2`` hook is executed in that case. This hook invokes the ``ENTRYPOINT_NAME``
+    on the smart contract ``INDEX`` with the ``OnReceivingCis2Params`` parameters
```

Comment:
I think we should mention that the data field is a list of bytes.

Reply:
I wrote the following:
The ``DATA_STRING`` is relevant because wCCD is sent to a smart contract and it is a string of bytes (string with lowercase hex).